### ISSUE TITLE
offstrategy compaction does not always run in maintenance scheduling group

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylla#10151]
It was assumed that offstrategy compaction is always triggered by streaming/repair
where it would inherit the caller's scheduling group.

However, offstrategy is triggered by a timer via `table::_off_strategy_trigger` so I don't see
how the expiration of this timer will inherit anything from streaming/repair.

Also, since d309a867088dd3f43c6e4bd4ebef00e543dccebd, offstrategy compaction
may be triggered by the api where it will run in the default scheduling group.

The bottom line is that the compaction manager needs to explicitly perform offstrategy compaction
in the maintenance scheduling group similar to `perform_sstable_scrub_validate_mode`.

**Comment by avikivity** [comment]
Backported to 4.6. Although just a performance fix, it can have major impact.
